src/hono.ts

```diff
@@ -9,7 +9,11 @@ export class Hono<
   S extends Schema = {},
   BasePath extends string = '/'
 > extends HonoBase<E, S, BasePath> {
-  constructor(init: Partial<Pick<Hono, 'router' | 'getPath'> & { strict: boolean }> = {}) {
+  constructor(
+    init: Exclude<ConstructorParameters<typeof HonoBase>[0], 'basePath'> & {
+      basePath?: BasePath
+    } = {}
+  ) {
```

Comment:
I've been wondering, why not just make the constructor params into a new type? since it's the same one with the other preset

Reply:
I also would like do that, but can't set the `BasePath` type (line:10) to `basePath`.

Reply:
why not make the type into some sort of generic instead? so it can be like `type HonoContructor<T extends string>`

Reply:
Do you mean like this?

```ts
type GetPath<E extends Env> = (request: Request, options?: { env?: E['Bindings'] }) => string

type HonoOptions<E extends Env, BasePath extends string> = {
  strict?: boolean
  basePath?: BasePath
  router?: Router<H>
  getPath?: GetPath<E>
}

export class Hono<
  E extends Env = Env,
  S extends Schema = {},
  BasePath extends string = '/'
> extends HonoBase<E, S, BasePath> {
  constructor(init: HonoOptions<E, BasePath> = {}) {
    super(init)
    this.router =
      init.router ??
      new SmartRouter({
        routers: [new RegExpRouter(), new TrieRouter()],
      })
  }
}
```

Reply:
yup, I believe this way anyone who want to contribute also can know that if they change the base preset, they also need to change other preset as well since they have a same type, etc.

Reply:
I see! I'll consider it and may fix it. Thanks!